parsers/PH.py

```diff
@@ -0,0 +1,696 @@
+    df["mode"] = df["resource_name"].apply(
+        lambda x: RESOURCE_NAME_TO_MODE[x] if x in RESOURCE_NAME_TO_MODE else "unknown"
+    )
+    # Fill a list of all unknown resources
+    unknown_resources = df[df["mode"] == "unknown"]["resource_name"].unique()
+    logger.warning(
+        f"PH - production: Unknown resources found: {unknown_resources}. Please report this issue to the Electricity Maps team"
+    )
+    return df
+
+
+def aggregate_per_datetime_mode(df: pd.DataFrame) -> pd.DataFrame:
+    return df.groupby(["datetime", "mode"]).sum(numeric_only=True).reset_index()
+
+
+def pivot_per_mode(df: pd.DataFrame) -> pd.DataFrame:
+    df = df.copy()
+    df = df.pivot(index="datetime", columns="mode")
+    # Flatten columns and make them "production.{mode}"
+    df.columns = df.columns.to_series().str.join(".")
+    # Handle storage
+    if "production.hydro_storage" in df.columns:
+        df = df.rename(columns={"production.hydro_storage": "storage.hydro"})
+        df["storage.hydro"] *= -1
+    return df
+
+
+def download_exchange_market_reports_items(
+    session: Session, reports_items: List[MarketReportsItem], logger: Logger
+) -> pd.DataFrame:
+    _all_items_df = []
+    for reports_item in reports_items:
+        logger.debug(f"Downloading market reports for {reports_item.filename}")
+        r: Response = session.get(reports_item.link)
+        _df = pd.read_csv(r.url)
+        _df = _df[:-1]
+        _df = convert_column_to_datetime(_df, "RUN_TIME")
+        _df["zone_key"] = _df["HVDC_NAME"].apply(
+            lambda x: EXCHANGE_KEY_MAPPING[x]["zone_key"]
+        )
+        _df["net_flow"] = _df.apply(
+            lambda x: EXCHANGE_KEY_MAPPING[x["HVDC_NAME"]]["flow"] * x["FLOW_FROM"],
+            axis=1,
+        )
+        _df = _df[["datetime", "zone_key", "net_flow"]]
+        _all_items_df.append(_df)
+
+    df = pd.concat(_all_items_df, ignore_index=True)
+    return df
+
+
+def convert_column_to_datetime(df: pd.DataFrame, datetime_column: str) -> pd.DataFrame:
+    """convert datetime column to datetime format"""
+    dt1 = pd.to_datetime(
+        df[datetime_column],
+        format="%m/%d/%Y %I:%M:%S %p",
+        errors="coerce",
+    )
+    dt2 = pd.to_datetime(df[datetime_column], format="%m/%d/%Y", errors="coerce")
+    df["datetime"] = dt1.combine_first(dt2)
+    return df
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production(
+    zone_key: ZoneKey = "PH-LU",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    reports_items = get_all_market_reports_items("production", logger)
+    reports_items = filter_reports_items(reports_items, target_datetime)
+
+    if len(reports_items) == 0:
+        raise ParserException(
+            parser="PH.py",
+            zone_key=zone_key,
+            message=f"{zone_key}: No production data available for {target_datetime.date()} ",
```

Comment:
target_datetime can be None here so pylance is flagging that we can't call .date() on a None object.